calcom/cal.com
# Add Google cal extneral calendar id to booking reference

## What does this PR do?



On Datadog I noticed a majority of the errors are coming from Google calendar with the error that the event is not found. @andreaestefania12 noticed that when you change the destination calendar when you try to update or delete a booking on the previous selected booking it would throw an error since it's searching for the booking on the currently selected calendar. Also mentioned by @zomars on #1501.

This PR builds off of #2551 by adding the external calendar id to `bookingReference`.

Fixes #2009 #2434 



## Type of change





## How should this be tested?



- Ensure that you have a Google calendar with two different calendars
- Create a booking
- Go to "Installed Apps" and change the destination calendar to a different one in your Google calendar
- Go to the "Bookings" page and update the booking that was created on the previously selected calendar
- Delete the booking
    - The changes should be reflected on your Google calendar

## Checklist




- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
